src/test/java/teammates/test/cases/webapi/GetFeedbackSessionsActionTest.java

```diff
@@ -20,16 +32,249 @@ protected String getRequestMethod() {
         return GET;
     }
 
-    @Test
+    @Override
+    protected void prepareTestData() {
+        sessionsInCourse1 = new ArrayList<>();
+        sessionsInCourse1.add(typicalBundle.feedbackSessions.get("session2InCourse1"));
+        sessionsInCourse1.add(typicalBundle.feedbackSessions.get("gracePeriodSession"));
+        sessionsInCourse1.add(typicalBundle.feedbackSessions.get("closedSession"));
+        sessionsInCourse1.add(typicalBundle.feedbackSessions.get("empty.session"));
+        sessionsInCourse1.add(typicalBundle.feedbackSessions.get("awaiting.session"));
+
+        sessionsInCourse2 = new ArrayList<>();
+        sessionsInCourse2.add(typicalBundle.feedbackSessions.get("session1InCourse2"));
+        sessionsInCourse2.add(typicalBundle.feedbackSessions.get("session2InCourse2"));
+
+        FeedbackSessionAttributes session1InCourse1 = typicalBundle.feedbackSessions.get("session1InCourse1");
+        session1InCourse1.setDeletedTime(Instant.now());
+
+        // Make student2InCourse2 and instructor1OfCourse1 belong to the same account.
+        StudentAttributes student2InCourse2 = typicalBundle.students.get("student2InCourse2");
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        student2InCourse2.googleId = instructor1OfCourse1.getGoogleId();
+
+        removeAndRestoreDataBundle(typicalBundle);
+    }
+
     @Override
     protected void testExecute() throws Exception {
-        // TODO
+        // see individual tests
+    }
+
+    @Test
+    protected void testExecute_asInstructorWithCourseId_shouldReturnAllSessionsForCourse() {
+        InstructorAttributes instructor2OfCourse1 = typicalBundle.instructors.get("instructor2OfCourse1");
+        loginAsInstructor(instructor2OfCourse1.googleId);
+
+        // The presence of IS_IN_RECYCLE_BIN flag is just used to indicate perform this request as instructor.
```

Comment:
This is a sign that there are some problems with the design.

Previously I thought we could distinguish the request from instructor/student by just `IS_IN_RECYCLE_BIN`. It seems that it is not true anymore.

Let's see if we can use entityType (use GetCoursesAction as a reference)?

Reply:
So you should remove the comment?